docs/rfd.md

```diff
@@ -0,0 +1,478 @@
+  // Memory soft limit defined in bytes. Processes exceeding the limit are
+  // not OOM-killed, but may be throttled by the kernel.
+  optional int64 softLimit = 1;
+  // Memory limit defined in bytes. If exceeded, the process will be OOM-killed.
+  optional int64 limit = 2;
+}
+
+message IODeviceLimits {
+  // Limits for individual devices
+  repeated DeviceLimit devices = 1;
+}
+
+message DeviceLimit {
+  // Device id (major:minor)
+  string deviceId = 1;
+  // Limits for the device
+  IOLimits limits = 2;
+}
+
+message IOLimits {
+  // Limit for read operations in bytes per second
+  optional int64 readBps = 2;
+  // Limit for write operations in bytes per second
+  optional int64 writeBps = 3;
+  // Limit for read operations in IOPS
+  optional int64 readIops = 4;
+  // Limit for write operations in IOPS
+  optional int64 writeIops = 5;
+}
+
+message OutputRequest {
+  JobId id     = 1;
+  bool  follow = 2;
+}
+
+```
+
+Authorization uses a very simple RBAC system. The RBAC configuration is defined as follows:
+
+```protobuf
+syntax = "proto3";
+
+package rbac.v1;
+
+option go_package = "github.com/kralicky/jobserver/pkg/apis/rbac/v1";
+
+// Describes a complete RBAC configuration.
+message Config {
+  // A list of available roles.
+  repeated Roles roles = 2;
+  // A list of available role bindings.
+  repeated RoleBindings roleBindings = 3;
+}
+
+// Describes a role that allows access to methods within a service.
+message Roles {
+  // An arbitrary unique identifier for the role.
+  string id = 1;
+  // The service name to which the role applies. Should be qualified to
+  // the full package name of the service, not including '/' separators.
+  // For example, `service Foo` in `package bar.baz` should be "bar.baz.Foo".
+  string service = 2;
+  // A list of methods that the role allows access to. The method names must
+  // not be qualified with the service name. All methods must exist in the
+  // named service. For example, `rpc Bar` in `service Foo` should be "Bar".
+  repeated string allowedMethods = 3;
+}
+
+// Describes a role binding, associating a single role with one or more subjects.
+message RoleBindings {
+  // An arbitrary unique identifier for the role binding.
+  string id = 1;
+  // An existing role id.
+  string roleId = 2;
+  // A list of subjects that the role applies to. A subject generally refers to
+  // a named user or group, but the actual meaning is left as an implementation
+  // detail for the server, as it largely depends on how its authentication
+  // system is configured.
+  repeated string subjects = 3;
+}
+
+```
+
+The RBAC configuration will be simply loaded from a config file on server startup. An example config file might look like:
+
+```yaml
+roles:
+  - id: adminRole
+    service: job.v1.Job
+    allowedMethods:
+      - Start
+      - Stop
+      - Status
+      - List
+      - Output
+roleBindings:
+  - id: adminRoleBinding
+    roleId: adminRole
+    subjects:
+      - admin
```

Comment:
How will the server determine the client's permissions? You mention X.509 certificate common name below in authz section - is that what "subjects" here is referring to?